src/main/java/uk/gov/companieshouse/api/accounts/controller/CompanyAccountController.java

```diff
@@ -19,7 +21,15 @@
 
     @PostMapping(value = "/transactions/{transactionId}/company-accounts",
             produces = MediaType.APPLICATION_JSON_VALUE)
-    public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount) {
-        return ResponseEntity.status(HttpStatus.CREATED).body(companyAccountService.save(companyAccount));
+    public ResponseEntity createCompanyAccount(@Valid @RequestBody CompanyAccount companyAccount)
+            throws NoSuchAlgorithmException {
+        //TODO accountNumber should be retrieved from the Transaction after it becomes available via the interceptor
+        SecureRandom random = new SecureRandom();
+        byte[] bytes = new byte[6];
+        random.nextBytes(bytes);
+        String accountNumber = new String(bytes);
```

Comment:
Has a story been raised to make sure this isn't forgotten when the transaction is available via the interceptor?

Reply:
Yes SFA-526

Reply:
I don't think the nosuchalgorithm should be thrown this far up... it should be handled in the generation of the id which should be done in the helper repo

Reply:
Additionally, this id generation for company accounts is already done and will be the same as abridged.